src/aliceVision/sfm/bundle/BundleAdjustmentCeres.cpp

```diff
@@ -592,15 +662,22 @@ void BundleAdjustmentCeres::addLandmarksToProblem(const sfmData::SfMData& sfmDat
 
             // needed parameters to create a residual block (K, pose)
             double* poseBlockPtr = _posesBlocks.at(view.getPoseId()).data();
-            double* intrinsicBlockPtr = _intrinsicsBlocks.at(view.getIntrinsicId()).data();
-            const std::shared_ptr<IntrinsicBase> intrinsic = _intrinsicObjects[view.getIntrinsicId()];
+            double* intrinsicBlockPtr = _intrinsicsBlocks.at(intrinsicId).data();
+            const std::shared_ptr<IntrinsicBase> intrinsic = _intrinsicObjects[intrinsicId];
+
+            double * distortionBlockPtr = fakeDistortionBlockPtr;
+            if (_distortionsBlocks.find(intrinsicId) != _distortionsBlocks.end())
+            {
+                distortionBlockPtr = _distortionsBlocks.at(intrinsicId).data();
+            }
 
             // apply a specific parameter ordering:
             if (_ceresOptions.useParametersOrdering)
             {
                 _linearSolverOrdering.AddElementToGroup(landmarkBlockPtr, 0);
                 _linearSolverOrdering.AddElementToGroup(poseBlockPtr, 1);
                 _linearSolverOrdering.AddElementToGroup(intrinsicBlockPtr, 2);
+                _linearSolverOrdering.AddElementToGroup(distortionBlockPtr, 2);
```

Comment:
Shouldn't this be
```suggestion
                _linearSolverOrdering.AddElementToGroup(distortionBlockPtr, 3);
```
?

Reply:
_linearSolverOrdering is telling the solver in which order the  variables are solved and subtituted (using Schur complement). By using the same group for both distortion and intrinsic, i'm keeping the same solver logic than before. So the answer, for me, is no, it shouldn't be 3.